src/stan/model/prob_grad.hpp

```diff
@@ -2,46 +2,91 @@
 #define STAN_MODEL_PROB_GRAD_HPP
 
 #include <cstddef>
+#include <sstream>
+#include <stdexcept>
+#include <string>
 #include <utility>
 #include <vector>
 
 namespace stan {
 
   namespace model {
 
+
     /**
-     * The <code>prob_grad</code> class represents the basic parameter
-     * holders for a model.  The command <code>bin/stanc</code> builds
-     * Models extending this base helper class.
+     * Base class for models, holding the basic parameter sizes and
+     * ranges for integer parameters.
      */
     class prob_grad {
-    protected:
+     protected:
+      // TODO(carpenter): roll into model_base; remove int members/vars
```

Comment:
This is not related to this PR, but rather to my ingorance... but why is the number of parameters in a model not declared `const`? These never change.